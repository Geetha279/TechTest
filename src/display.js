import React from 'react';
import './App.css';
import { connect } from 'react-redux';

export class Display extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            name: ''
        }

    };
    // handleChange = (e) => {
    //    this.addName(e.target.value)
    // };

    render() {

        const { pin } = this.props;
        console.log("this.props", this.props);
        return (

            <React.Fragment>
                <div className="Display">
                    {pin.length > 0 ? "" : <p className="pStyle"> Pins are empty </p>}
                    <div><table><tbody>{pin.length > 0 ? pin.map((item, index) => <tr key={index} style={{ height: '29px' }}><td key={index + 1}><input className="inputStyle" placeholder={item.name} type="text" onChange={(e) => this.props.addName(e.target.value, item.id)} /></td>{item.value.map((item, index) => <td className="tableDisplayStyle" key={index + 3}>{item}</td>)}<td key={index + 5}><button onClick={() => this.props.deletePin(this.state, item.id)} className="DeleteButton">DELETE</button></td></tr>) : <tr></tr>}</tbody></table></div></div>
            </React.Fragment>
        );
    }
}
const mapDispatchToProps = dispatch => {
    console.log("Display  dispatch methods");
    return {
        addName:(name,id)=>dispatch({
            type:"updatename",
             name:name,
             id:id
            }),
        deletePin: (response, id) => dispatch({
            type: "delete",
            id: id,
            result: response
        })

    }
}


const mapStateToProps = state => {
    console.log("home page mapstatetoprops ", state.pinArray);

    return {

        pin: state.pinArray,
        msg: state.name,
        id: state.id

    }
}


export default connect(mapStateToProps, mapDispatchToProps)(Display);



