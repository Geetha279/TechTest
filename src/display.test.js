import React from 'react';
import { shallow, configure } from 'enzyme';
import { Display } from './display';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Generate component', () => {
    it('check div', () => {
        const props = {
            pin: [{
                id: "sai",
                value: [1221, 3476, 5690, 2710, 2435]
            }, {
                id: "pratap",
                value: [1223, 3478, 5650, 2760, 2835]
            }, {
                id: "reddy",
                value: [1261, 4476, 7690, 2810, 1435]
            },
            ]
        }
        const wrapper = shallow(<Display {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

});




