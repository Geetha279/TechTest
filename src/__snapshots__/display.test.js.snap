// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate component check div 1`] = `ShallowWrapper {}`;
